vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -334,103 +334,107 @@ bool trans_map_1d_amr(const dccrg::Dccrg<spatial_cell::SpatialCell,dccrg::Cartes
+               // Transpose and copy block data from cells to source buffer
+               Vec* blockDataSource = blockDataBuffer.data() + start*WID3/VECL;
+               Realf** pencilBlockData = cellBlockData.data() + start;
+               copy_trans_block_data_amr(pencilBlockData, L, blockDataSource, vcell_transpose, popID);
             }
-            if(nonEmptyBlocks == 0) {
-               continue;
-            }
-            pencilBlocksCount.at(pencili) = nonEmptyBlocks;
-            // Transpose and copy block data from cells to source buffer
-            Vec* blockDataSource = blockDataBuffer.data() + start*WID3/VECL;
-            Realf** pencilBlockData = cellBlockData.data() + start;
-            copy_trans_block_data_amr(pencilBlockData, L, blockDataSource,
-                                      vcell_transpose, popID);
-         }
-         loadTimer.stop();
-
-         phiprof::Timer memsetTimer {memsetTimerId};
-         // reset blocks in all non-sysboundary neighbor spatial cells for this block id
-         for (CellID target_cell_id: DimensionTargetCells[dimension]) {
-            SpatialCell* target_cell = mpiGrid[target_cell_id];
-            if (target_cell) {
-               // Get local velocity block id
-               const vmesh::LocalID blockLID = target_cell->get_velocity_block_local_id(blockGID, popID);
-               // Check for invalid block id
-               if (blockLID != vmesh::VelocityMesh::invalidLocalID()) {
-                  // Get a pointer to the block data
-                  Realf* blockData = target_cell->get_data(blockLID, popID);
-                  memset(blockData, 0, WID3*sizeof(Realf));
+            loadTimer.stop();
+
+            phiprof::Timer memsetTimer {memsetTimerId};
+            // reset blocks in all non-sysboundary neighbor spatial cells for this block id
+            for (CellID target_cell_id: DimensionPencils[dimension].binsCells[DimensionPencils[dimension].activeBins[nBin]]) {
+               SpatialCell* target_cell = mpiGrid[target_cell_id];
+               if (target_cell) {
+                  // Get local velocity block id
+                  const vmesh::LocalID blockLID = target_cell->get_velocity_block_local_id(blockGID, popID);
+                  // Check for invalid block id
+                  if (blockLID != vmesh::VelocityMesh::invalidLocalID()) {
+                     // Get a pointer to the block data
+                     Realf* blockData = target_cell->get_data(blockLID, popID);
+                     memset(blockData, 0, WID3*sizeof(Realf));
+                  }
                }
             }
-         }
-         memsetTimer.stop();
+            memsetTimer.stop();
+
+            phiprof::Timer propagateTimer {propagateTimerId};
+            for (uint pencili : DimensionPencils[dimension].binsPencils[DimensionPencils[dimension].activeBins[nBin]]) {
+               // Skip pencils without blocks
+               if (pencilBlocksCount.at(pencili) == 0)
```

Comment:
Please, let's not explicitly remove braces from if-clauses :) even if it's "trivial".